### PR TITLE
Add baseline implementation of HTTP client for inference server

### PR DESCRIPTION
**Pull request by PawelPeczek-Roboflow**
# Description
* created a python client for `inference` HTTP API
* extended server to provide detailed informations about  models
* client works against inference HTTP server running locally
* client should also work against hosted roboflow models - but only in `v0` mode
* 
## Type of change

Please delete options that are not relevant.

-   [x] New feature (non-breaking change which adds functionality)
-   [x] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?
Tests suits provided, also tested against real input

## Docs

-   [x] Docs updated? What were the changes: added docs for whole HTTP client

### CODE REVIEW

**Comment by PawelPeczek-Roboflow** [comment]
Conceptual usage patterns:

1) Endpoints to control server
```python
API_KEY = "<api_key>"

from clients.http.client import InferenceHTTPClient
from clients.http.entities import InferenceConfiguration, ModelType

CLIENT.get_server_info()
>>> ServerInfo(name='Roboflow Inference Server', version='0.8.8', uuid='DV9pSZ')


## UPFRONT LOADING OF MODEL - flow

CLIENT.load_model(
    model_id="wood-log-counting/7",
    model_type=ModelType.OBJECT_DETECTION,
    set_as_default=True,
)
CLIENT.load_model(
    model_id="leaf-t9v7t/1",
    model_type=ModelType.CLASSIFICATION,
)
CLIENT.load_model(
    model_id="corrosion-instance-segmentation-sfcpc/3",
    model_type=ModelType.INSTANCE_SEGMENTATION,
)
CLIENT.list_loaded_models()
>>> RegisteredModels(model_ids=['leaf-t9v7t/1', 'wood-log-counting/7', 'corrosion-instance-segmentation-sfcpc/3'])
```

2) Usage new API
```python
CLIENT.predict(
    "/Users/ppeczek/Documents/assets/example_images/inference/logs.jpg",
)
>>> dict response from API
```

3) Dynamic selection of models to execute specific request:
```python
CLIENT.predict(
    "/Users/ppeczek/Documents/assets/example_images/inference/corrosion.jpg",
    model_id="corrosion-instance-segmentation-sfcpc/3",
    model_type=ModelType.INSTANCE_SEGMENTATION,
)
```

4) static selection of model
```python
CLIENT = CLIENT.use_model(model_id="leaf-t9v7t/1", model_type=ModelType.CLASSIFICATION)

## and then without pointing model to use
CLIENT.predict(
    "/Users/ppeczek/Documents/assets/example_images/inference/corrosion.jpg"
)
```

5) Using legacy API
```python
CLIENT = CLIENT.use_legacy_client()

# then everything works the same
CLIENT.predict(
    "/Users/ppeczek/Documents/assets/example_images/inference/corrosion.jpg"
)
```

6) configuring custom parameters
```python
CUSTOM_CONFIGURATION = InferenceConfiguration(
    confidence_threshold=0.999,
)
CLIENT = CLIENT.configure(CUSTOM_CONFIGURATION)
```

7) Ideas:
* context managers for model usage (to have `with CLIENT.use_legacy()` and `with CLIENT.use_model()`
* builders for `InferenceConfiguration` (to have InferenceConfiguration().set_threshold(0.9).another_option(...)`

8) issues:
* HTTP500 on new API for segmantation and image classification - seems like response cannot pass validation:
```
[end of stacktrace]
File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 249, in app
    |     content = await serialize_response(
    |   File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 138, in serialize_response
    |     raise ValidationError(errors, field.type_)
    | pydantic.error_wrappers.ValidationError: <exception str() failed>
```

File: .release/pypi/inference.clients.setup.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        path = [path]
+    requirements = []
+    for p in path:
+        with open(p) as fh:
+            requirements.extend([line.strip() for line in fh])
+    return requirements
+
+
+setuptools.setup(
+    name="inference-clients",
```
Comment (paulguerrie) [inline .release/pypi/inference.clients.setup.py:33]: can we change this to just `inference-client`

File: inference_clients/version.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+__version__ = "0.8.9"
```
Comment (paulguerrie) [inline inference_clients/version.py:1]: Let's not version this file. Let's let it just get copied in during the pypi publishing step.

File: .release/pypi/inference.clients.setup.py (hunk 1 above)
Comment (PawelPeczek-Roboflow) [reply .release/pypi/inference.clients.setup.py:33]: 👍

File: inference_clients/version.py (hunk 2 above)
Comment (PawelPeczek-Roboflow) [reply inference_clients/version.py:1]: ok

**Review by paulguerrie — APPROVED** [review]
Looks great!!

**Pull request merged**